File: src/cz/cuni/lf1/lge/ThunderSTORM/AboutPlugIn.java
package cz.cuni.lf1.lge.ThunderSTORM;

import cz.cuni.lf1.lge.ThunderSTORM.UI.GUI;
import cz.cuni.lf1.lge.ThunderSTORM.UI.Help;
import ij.IJ;
import ij.plugin.BrowserLauncher;
import ij.plugin.PlugIn;
import java.awt.Cursor;
import java.awt.Dialog;
import java.awt.Dimension;
import java.awt.Window;
import javax.swing.BorderFactory;
import javax.swing.JDialog;
import javax.swing.JScrollPane;
import javax.swing.event.HyperlinkEvent;
import javax.swing.event.HyperlinkListener;
import org.fit.cssbox.swingbox.BrowserPane;
import org.fit.cssbox.swingbox.SwingBoxEditorKit;

public class AboutPlugIn implements PlugIn {

    private static final String url = "resources/help/about.html";
    private static final int WINDOW_WIDTH = 600;
    private static final int WINDOW_HEIGHT = 600;

    @Override
    public void run(String arg) {
        GUI.setLookAndFeel();
        JDialog dialog = new JDialog(IJ.getInstance(), "About ThunderSTORM (" + ThunderSTORM.VERSION + ")");
        if(IJ.isJava17()) {
            dialog.setType(Window.Type.UTILITY);
        }
        dialog.setModalExclusionType(Dialog.ModalExclusionType.APPLICATION_EXCLUDE); //for use within modal dialog
        final BrowserPane htmlBrowser = new BrowserPane();
        htmlBrowser.setEditorKit(new SwingBoxEditorKit());
        htmlBrowser.setBorder(BorderFactory.createEmptyBorder());
        htmlBrowser.addHyperlinkListener(new HyperlinkListener() {
            @Override
            public void hyperlinkUpdate(HyperlinkEvent e) {
                if(e.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
                    try {
                        if("jar".equals(e.getURL().getProtocol())) {
                            htmlBrowser.setPage(e.getURL());
                        } else {
                            BrowserLauncher.openURL(e.getURL().toString());
                        }
                    } catch(Exception ex) {
                        IJ.handleException(ex);
                    }
                } else if(e.getEventType() == HyperlinkEvent.EventType.ENTERED) {
                    htmlBrowser.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
                } else if(e.getEventType() == HyperlinkEvent.EventType.EXITED) {
                    htmlBrowser.setCursor(Cursor.getDefaultCursor());
                }

            }
        });
        JScrollPane scrollPane = new JScrollPane(htmlBrowser);
        scrollPane.setPreferredSize(new Dimension(WINDOW_WIDTH, WINDOW_HEIGHT));
        dialog.getContentPane().add(scrollPane);

        try {
            htmlBrowser.setPage(Help.getUrl(url));
        } catch(Exception e) {
            htmlBrowser.setText("Could not load resource file.");
        }

        dialog.pack();
        dialog.setLocationRelativeTo(null);
        dialog.setVisible(true);
    }
}

File: src/main/java/cz/cuni/lf1/lge/ThunderSTORM/AboutPlugIn.kt
package cz.cuni.lf1.lge.ThunderSTORM

import cz.cuni.lf1.lge.ThunderSTORM.UI.GUI
import cz.cuni.lf1.lge.ThunderSTORM.UI.Help
import cz.cuni.lf1.lge.ThunderSTORM.UI.HelpButton
import ij.IJ
import ij.plugin.BrowserLauncher
import ij.plugin.PlugIn
import java.awt.Cursor
import java.awt.Dialog
import java.awt.Dimension
import java.awt.Window
import javax.swing.BorderFactory
import javax.swing.JDialog
import javax.swing.JScrollPane
import javax.swing.event.HyperlinkEvent

private const val URL = "resources/help/about.html"
private const val WINDOW_WIDTH = 600
private const val WINDOW_HEIGHT = 600

class AboutPlugIn : PlugIn {
    override fun run(arg: String) {
        try {
            GUI.setLookAndFeel()
            val dialog = JDialog(IJ.getInstance(), "About ThunderSTORM (" + ThunderSTORM.VERSION + ")")
            if(IJ.isJava17()) {
                dialog.type = Window.Type.UTILITY
            }
            dialog.modalExclusionType = Dialog.ModalExclusionType.APPLICATION_EXCLUDE //for use within modal dialog
            val htmlBrowser = HelpButton.createEditorUsingOurClassLoader()
            htmlBrowser.border = BorderFactory.createEmptyBorder()
            htmlBrowser.addHyperlinkListener({ e ->
                    if(e.eventType == HyperlinkEvent.EventType.ACTIVATED) {
                        try {
                            if("jar".equals(e.url.protocol)) {
                                htmlBrowser.page = e.url
                            } else {
                                BrowserLauncher.openURL(e.url.toString())
                            }
                        } catch(ex: Exception) {
                            IJ.handleException(ex)
                        }
                    } else if(e.eventType == HyperlinkEvent.EventType.ENTERED) {
                        htmlBrowser.cursor = Cursor.getPredefinedCursor(Cursor.HAND_CURSOR)
                    } else if(e.eventType == HyperlinkEvent.EventType.EXITED) {
                        htmlBrowser.cursor = Cursor.getDefaultCursor()
                    }
            })
            val scrollPane = JScrollPane(htmlBrowser)
            scrollPane.preferredSize = Dimension(WINDOW_WIDTH, WINDOW_HEIGHT)
            dialog.contentPane.add(scrollPane)

            htmlBrowser.page = Help.getUrl(URL)

            dialog.pack()
            dialog.setLocationRelativeTo(null)
            dialog.isVisible = true
        } catch(e: Exception) {
            IJ.handleException(e)
        }
    }
}
